dagstore_control.go

```diff
@@ -133,6 +144,7 @@ func (d *DAGStore) control() {
 			s.wAcquire = s.wAcquire[:0]
 
 		case OpShardAcquire:
+			s.lastAccessedAt = time.Now()
```

Comment:
Storing the timestamp here leaks the GC abstraction. Instead, make a GC abstraction that you `NotifyAccessed(Shard)`. That GC abstraction can be a noop (if no automated GC is enabled), it could be an LRU cache, an LFU cache, an ARC cache, etc. But this approach is not right as it leaks the LRU strategy into the shard's data model.